cpp/src/strings/regex/regex.inl

```diff
@@ -40,95 +33,101 @@ namespace detail {
  * reflected here. The regexec function updates and manages this state data.
  */
 struct alignas(8) relist {
-  int16_t size{};
-  int16_t listsize{};
-  int32_t reserved;
-  int2* ranges{};       // pair per instruction
-  int16_t* inst_ids{};  // one per instruction
-  u_char* mask{};       // bit per instruction
-
-  CUDF_HOST_DEVICE inline static int32_t data_size_for(int32_t insts)
+  /**
+   * @brief Compute the memory size for the state data.
+   */
+  constexpr inline static std::size_t data_size_for(int32_t insts)
   {
     return ((sizeof(ranges[0]) + sizeof(inst_ids[0])) * insts) + ((insts + 7) / 8);
   }
 
-  CUDF_HOST_DEVICE inline static int32_t alloc_size(int32_t insts)
+  /**
+   * @brief Compute the aligned memory allocation size.
+   */
+  constexpr inline static std::size_t alloc_size(int32_t insts)
   {
-    int32_t size = sizeof(relist);
-    size += data_size_for(insts);
-    size = ((size + 7) / 8) * 8;  // align it too
-    return size;
+    return cudf::util::round_up_unsafe<size_t>(data_size_for(insts) + sizeof(relist),
+                                               sizeof(ranges[0]));
   }
 
-  CUDF_HOST_DEVICE inline relist() {}
+  struct alignas(16) restate {
+    int2 range;
+    int32_t inst_id;
+    int32_t reserved;
+  };
 
-  CUDF_HOST_DEVICE inline relist(int16_t insts, u_char* data = nullptr) : listsize(insts)
+  __device__ __forceinline__ relist(int16_t insts, u_char* data = nullptr)
+    : masksize((insts + 7) / 8)
```

Comment:
another opportunity for round up

Reply:
I'm not sure how to use `round_up` here. 
This is converting number of bits to a whole byte: `0-7 -> 0, 8-15 -> 1, 16-23 -> 2`.

Reply:
Isn't that what this does? https://github.com/rapidsai/cudf/blob/c1638869116aae2c6dde6024394279a2fb79e685/cpp/include/cudf/detail/utilities/integer_utils.hpp#L110-L113

Reply:
Sorry, I did not see that one. Thanks.